README.md

```diff
@@ -1,58 +1,106 @@
+## Motivation
 
-## Code Example
-Show what the library does as concisely as possible, developers should be able to figure out **how** your project solves their problem by looking at the code example. Make sure the API you are showing off is obvious, and that your code is short and concise.
+- Learn how to design test for each operation.
+- Learn how to route user request to handler.
+- Learn how to handle error of user request.
+
+## What will learn from this exercise
+
+### API flow of HTML request:
+
+| API flow           | Purpose                                   |
+| ------------------ | ----------------------------------------- |
+| index.js           | process config env                        |
+| src/app.js         | process routers                           |
+| src/routes/\*      | process get/post/put/patch controllers    |
+| src/controllers/\* | process back-end controllers and response |
+
+### Database setup for each test:
+
+To ensure tests have a fresh database each time they run.
+
+```
+// package.json
+ ...
+ "scripts": {
+    "migrate": "node scripts/migrate.js",
+    "prestart": "node scripts/create-database.js && npm run migrate",
+    "start": "nodemon -r dotenv/config index.js",
+    "pretest": "node scripts/create-database.js test && npm run migrate test",
+    "test": "mocha tests/**/*.js --exit --recursive --timeout 60000 --file ./scripts/test-setup.js",
+    "posttest": "node scripts/drop-database.js test"
+  },
+  ...
+```
+
+### A top-level directory layout
+
+    .
+    ├── migrations         # Setup databases before each test
+    ├── scripts            # Scripts for migrations
+    ├── src
+    │      ├── routes      # Source of routers
+    │      ├── controllers # Source of request handlers
+    │      └── db          # define query of pg
+    ├── tests              # test files for requests
+    └── README.md
 
 ## Installation
-Provide step by step series of examples and explanations about how to get a development env running.
+
+```
+$ git clone https://github.com/DavidCheungTW/music-library.git
+$ cd music-library
+$ npm install
+```
 
 ## API Reference
 
-Depending on the size of the project, if it is small and simple enough the reference docs can be added to the README. For medium size to larger projects it is important to at least provide a link to where the API reference docs live.
+```
+// package.json
+...
+  "devDependencies": {
+    "chai": "^4.3.7",
+    "dotenv": "^16.0.3",
+    "mocha": "^10.2.0",
+    "nodemon": "^2.0.20",
+    "supertest": "^6.3.3"
+  },
+  ...
+```
 
 ## Tests
-Describe and show how to run the tests with code examples.
+
+```
+open terminal and go to directory 'music-library', then
+$ npm test
+```
 
 ## How to use?
-If people like your project they’ll want to learn how they can use it. To do so include step by step guide to use your project.
+
+From this exercise, you may learn how to handle request for CRUD operation to relational database.
```

Comment:
This isnt how to use the app, its what you gained from creating it :) 
This is where you would explain to a user who has never seen it before how they can go about using the app (IE is it on the command line? Throughout a GUI? What can they expect?)